src/backend/oneapi/kernel/index.hpp

```diff
@@ -133,11 +133,14 @@ void index(Param<T> out, Param<T> in, IndexKernelParam& p,
     blocks[0] *= threads[0];
 
     sycl::nd_range<3> marange(blocks, threads);
+    sycl::buffer<uint> *idxArrs_get[4];
```

Comment:
Wouldn't this go out of scope when you call submit?

Reply:
This is captured by the lambda function which is passed as an argument to submit.